spec/game_spec.rb

```diff
@@ -1,17 +1,125 @@
 RSpec.describe Game do
-  context "#initialize" do
-    it "check class of difficulty"
-    it "check class of hints_total"
-    it "check class of secret_code"
-    it "check class of hint_clone_scode"
+  let(:game) { described_class.new(Console::DIFF[:medium]) }
+  let(:console_game) { Console_game.new('Sergey', Console::DIFF[:medium]) }
+  let (:difficulty) { game.difficulty }
+  let (:secret_code) { game.secret_code }
+  let (:hint_clone_scode) { game.hint_clone_scode }
+
+  context 'when the game started and datas move to processor' do
+    it 'when the variable `difficulty` is exist', positive: true do
+      expect(difficulty.class).to eq(Hash)
+    end
+
+    it 'when the variable `secret_code` is exist', positive: true do
+      expect(secret_code.class).to eq(Array)
+    end
+
+    it 'when the variable `hint_clone_scode` is exist', positive: true do
+      expect(hint_clone_scode.class).to eq(Array)
+    end
+
+    it 'when the digit of `secret_code` is correct', positive: true do
+      expect(secret_code.size).to eq(Game::DIGIT)
+    end
+
+    it 'when the digit of `secret_code` equal `hint_clone_scode` at the start', positive: true do
+      expect(secret_code.size == hint_clone_scode.size).to eq(true)
+    end
+
+    it 'when the sequence of `secret_code` is not same as `hint_clone_scode`  at the start', positive: true do
+      expect(hint_clone_scode).not_to eq(secret_code)
+    end
   end
-  context "comparing user input " do
-    it "check class of user input"
-    it "when user inpute equales secret code"
+
+  context 'when user did a coorect input of number' do
+    it 'when secret code equles user input, return game status `win`' do
+      user_input = secret_code.join
+      allow(game).to receive(:compare).with(user_input).and_call_original
+      expect(game.compare(user_input)).to eq(Console::MESSAGE_GU[:win])
+    end
+    it 'when user input differen with  secret code, return  call plus-minus factoring' do
+      allow(game).to receive(:compare).with(NUMBER).and_call_original
+      expect(game.compare(NUMBER)).to receive(:plus_minus_factoring).with(NUMBER).once
```

Comment:
Failure/Error: expect(game.compare(NUMBER)).to receive(:plus_minus_factoring).with(NUMBER).once
     
       ("+-").plus_minus_factoring("1111")
           expected: 1 time with arguments: ("1111")
           received: 0 times